### DISCUSSION TITLE
ScriptReference/QualitySettings-vSyncCount

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#97] [Notes]
# ScriptReference/QualitySettings-vSyncCount

https://docs.unity3d.com/ScriptReference/QualitySettings-vSyncCount.html

**Comment by vertxxyz** [comment]
VSync will try to synchronize the framerate to divisions of the monitor's refresh rate.
#### Examples
**With a 60hz monitor**
```csharp
QualitySettings.vSyncCount = 0; // FPS capped by targetFrameRate
QualitySettings.vSyncCount = 1; // 60/1 = 60fps
QualitySettings.vSyncCount = 2; // 60/2 = 30fps
QualitySettings.vSyncCount = 3; // 60/3 = 20fps
QualitySettings.vSyncCount = 4; // 60/4 = 15fps
```
**With a 144hz monitor**
```csharp
QualitySettings.vSyncCount = 0; // FPS capped by targetFrameRate
QualitySettings.vSyncCount = 1; // 144/1 = 144fps
QualitySettings.vSyncCount = 2; // 144/2 = 72fps
QualitySettings.vSyncCount = 3; // 144/3 = 48fps
QualitySettings.vSyncCount = 4; // 144/4 = 36fps
```
---  
Screens with adaptive sync (AMD Freesync, Nvidia G-SYNC) adjust their refresh rate to fluctuating framerates between some limits. Ideally VSync is disabled, and a frame rate limit is applied using [`Application.targetFrameRate`](https://docs.unity3d.com/ScriptReference/Application-targetFrameRate.html) in these circumstances,  
Ideal graphics settings will show VSync and when disabled will also show a max frame rate setting.  
Without vsync a framerate cap well-running games will process frames extremely rapidly and you will get complaints from users as their computers become needlessly loud.

**Comment by joethephish** [comment]
Note that this value needs to be saved by hand between launches (such as with PlayerPrefs etc), and reapplied when loaded. This is unlike Screen.SetResolution which persists automatically.

If you're not sure what value to use for vSyncCount and don't want/need the game to draw at super high framerates, one option is to find the value that will give you the closest to 60fps. For example based on [this post](https://gamedev.stackexchange.com/questions/205365/how-to-limit-framerate-in-unity-without-causing-screentears), I use the following:

```
        if( enabled ) {
            var refreshRate = Screen.currentResolution.refreshRate;
            float vSyncFactor = refreshRate / 60.0f;
            QualitySettings.vSyncCount = Clamp(RoundToInt(vSyncFactor), 1, 4);
        }

        else {
            QualitySettings.vSyncCount = 0;
        }
```
